x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/AuditTrailService.java

```diff
@@ -16,24 +18,40 @@
 import org.elasticsearch.xpack.security.transport.filter.SecurityIpFilterRule;
 
 import java.net.InetAddress;
+import java.time.Duration;
+import java.time.Instant;
 import java.util.Collections;
 import java.util.List;
 
 public class AuditTrailService {
 
+    private static final Logger logger = LogManager.getLogger(AuditTrailService.class);
+
     private static final AuditTrail NOOP_AUDIT_TRAIL = new NoopAuditTrail();
     private final CompositeAuditTrail compositeAuditTrail;
     private final XPackLicenseState licenseState;
+    private final Duration minLogPeriod = Duration.ofMinutes(30);
+    // protected for tests
+    protected Instant lastLogInstant = Instant.EPOCH;
 
     public AuditTrailService(List<AuditTrail> auditTrails, XPackLicenseState licenseState) {
         this.compositeAuditTrail = new CompositeAuditTrail(Collections.unmodifiableList(auditTrails));
         this.licenseState = licenseState;
     }
 
     public AuditTrail get() {
-        if (compositeAuditTrail.isEmpty() == false &&
-            licenseState.isSecurityEnabled() && licenseState.checkFeature(Feature.SECURITY_AUDITING)) {
-            return compositeAuditTrail;
+        if (compositeAuditTrail.isEmpty() == false) {
+            if (licenseState.isSecurityEnabled() && licenseState.checkFeature(Feature.SECURITY_AUDITING)) {
+                return compositeAuditTrail;
+            } else {
+                Instant nowInstant = Instant.now();
+                if (lastLogInstant.plus(minLogPeriod).isBefore(nowInstant)) {
+                    lastLogInstant = nowInstant;
+                    logger.warn("Security auditing is DISABLED because the currently active license [" +
```

Comment:
Nit: should it be `Auditing logging ...` instead of `Security auditing ...`? It is true that auditing events are all security related, also this is how the feature Enum is named. But the feature is [documented](https://www.elastic.co/guide/en/elasticsearch/reference/current/enable-audit-logging.html) and [marketed](https://www.elastic.co/subscriptions) as "audit logging", which seems to be a better end-user oriented term.